mobile/examples/cc/fetch_client/fetch_client_main.cc

```diff
@@ -1,22 +1,112 @@
+  Envoy::EnvoyStreamCallbacks stream_callbacks;
+  stream_callbacks.on_headers_ = [](const Envoy::Http::ResponseHeaderMap& headers, bool /* end_stream */,
+                                    envoy_stream_intel intel) {
+    std::cerr << "Received headers on connection: " << intel.connection_id << "with headers:\n"
+              << headers << "\n";
+  };
+  stream_callbacks.on_data_ = [](const Envoy::Buffer::Instance& buffer, uint64_t length, bool end_stream,
+                                 envoy_stream_intel) {
+    std::string response_body(length, ' ');
+    buffer.copyOut(0, length, response_body.data());
+    std::cerr << response_body << "\n";
+    if (end_stream) {
+      std::cerr << "Received final data\n";
+    }
+  };
+  stream_callbacks.on_complete_ =
+      [request_finished, used_protocols](envoy_stream_intel, envoy_final_stream_intel final_intel) {
+        std::cerr << "Request finished after "
+                  << final_intel.stream_end_ms - final_intel.stream_start_ms << "ms\n";
+        used_protocols->push_back(static_cast<Envoy::Http::Protocol>(final_intel.upstream_protocol));
+        request_finished->Notify();
+      };
+  stream_callbacks.on_error_ = [request_finished, status](const Envoy::EnvoyError& error,
+                                                            envoy_stream_intel,
+                                                            envoy_final_stream_intel final_intel) {
+    *status = ENVOY_FAILURE;
+    std::cerr << "Request failed after " << final_intel.stream_end_ms - final_intel.stream_start_ms
+              << "ms with error message: " << error.message_ << "\n";
+    request_finished->Notify();
+  };
+  stream_callbacks.on_cancel_ = [request_finished](envoy_stream_intel,
+                                                    envoy_final_stream_intel final_intel) {
+    std::cerr << "Request cancelled after "
+              << final_intel.stream_end_ms - final_intel.stream_start_ms << "ms\n";
+    request_finished->Notify();
+  };
+  return stream_callbacks;
+}
+
 // Fetches each URL specified on the command line in series,
 // and prints the contents to standard out.
 int main(int argc, char** argv) {
-  Envoy::Fetch client;
   std::vector<absl::string_view> urls;
   // Start at 1 to skip the command name.
   for (int i = 1; i < argc; ++i) {
     urls.push_back(argv[i]);
   }
-  std::vector<Envoy::Http::Protocol> protocols;
-  client.fetch(urls, /* quic_hints=*/{}, /* protocols= */ protocols);
+
+  // Build and run engine.
+  absl::Notification engine_running;
+  Envoy::Platform::EngineSharedPtr engine = Envoy::Platform::EngineBuilder()
+      .setLogLevel(Envoy::Logger::Logger::trace)
+      .addRuntimeGuard("dns_cache_set_ip_version_to_remove", true)
+      .addRuntimeGuard("quic_no_tcp_delay", true)
+      .setOnEngineRunning([&engine_running]() { engine_running.Notify(); })
+      .build();
```

Comment:
I realized the tool used HTTP/3 after it learned about it with the multiple URL fetch, even if I don't call `.enableHttp3()`. That seems to be the default?